classes/class2/README.md

```diff
@@ -1 +1,630 @@
+## Objects
+
+Objects in JavaScript are simple key-value stores, which you may know from Python as **dictionaries**.
+
+We can create an empty object like this:
+
+```node
+> var obj = {}
+undefined
+```
+
+Properties can be added and accessed with dot syntax.
+
+```node
+> obj.color = 'blue'
+'blue'
+> obj.color
+'blue'
+```
+
+Bracket syntax lets us use variables as keys.
+
+```node
+> var colorKey = 'color'
+undefined
+> obj[colorKey]
+'blue'
+```
+
+Dot syntax is more concise than bracket syntax, so it's preferable unless bracket syntax is necessary (like if your key has a space, for some reason):
+
+```node
+> obj['another color'] = 'red'
+'red'
+> obj['another color']
+'red'
+> obj
+{ color: 'blue',
+'another color': 'red' }
+```
+
+Printing `obj` shows exactly how to declare objects that aren't empty. Objects can contain any variable type, including functions and other objects:
+
+```node
+> var obj = {
+... aBool: true,
+... aString: 'string',
+... aNum: 5.6,
+... aFunc: function() { return 6; },
+... anObj: { key: 'nesting is pretty great' }
+... }
+undefined
+```
+
+Properties can be treated like you might expect:
+
+```node
+> obj.aFunc
+[Function]
+> obj.aFunc()
+6
+> obj.anObj.key
+'nesting is pretty great'
+```
+
+JavaScript objects are versatile and ubiquitous — you'll see them everywhere. This brings us to the next section.
+
+### Arrays
+
+Arrays in JavaScript are also objects, which is why we covered objects first. Declare an array like this:
+
+```node
+> var arr = []
+undefined
+```
+
+Or a filled one like this:
+
+```node
+> var array = [1, 2, 3, 4]
+undefined
+```
+
+Arrays can contain any type, and need not contain variables of the same type.
+
+```node
+> var jsArray = [true, 'string', function() { return 'why would anyone do this'; }]
+undefined
+> jsArray[2]()
+'why would anyone do this'
+```
+
+### Array Methods
+
+#### Array length
+
+Retrieve the length of an array with `Array.length`:
+
+```node
+> array
+[ 1, 2, 3, 4 ]
+> array.length
+4
+```
+
+#### Appending and Prepending
+
+Use `push` to append an element to an array:
+
+```node
+> array.push(5)
+5
+> array
+[ 1, 2, 3, 4, 5 ]
+```
+
+And `pop` to remove the last element (`pop` returns the last element):
+
+```node
+> array.pop()
+5
+> array
+[ 1, 2, 3, 4 ]
+```
+
+Conversely, use `unshift` to prepend an element:
+
+```node
+> array.unshift(0)
+5
+> array
+[ 0, 1, 2, 3, 4 ]
+```
+
+And `shift` to remove the first element (`shift` returns the first element):
+
+```node
+> array.shift()
+0
+> array
+[ 1, 2, 3, 4 ]
+```
+
+#### Further Reading
+
+There are more built-in array methods, but we'll cover them later. JavaScript arrays generally act like Python arrays.
+
+Two good sources for documentation are [w3schools](http://www.w3schools.com/jsref/jsref_obj_array.asp) (easier to follow) and [Mozilla](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array) (official and more complete).
+
+### Weird JavaScript Array Behavior (Optional)
```

Comment:
How is anything in the readme optional? I think the non-homework parts of the readme should either be worth everyone reading, or not including. (And I think this is worth everyone taking the 90 seconds to read through it.)